### DISCUSSION TITLE
Alternatives for 1.x Data Binding Framework

### DISCUSSION THREAD

**Discussion by TW4177** [django/channels#1963] [Q&A]
We're up-reving our Django application to Channels 4.x, and this entails considerable refactoring of our existing business logic around websockets.  The Channels 2.x document [refers to removal of the binding framework](https://channels.readthedocs.io/en/2.x/one-to-two.html#removed-components) but I have not located any guidance on what to do with the business logic in our existing code derived from the deleted `channels.binding` modules, or how to make it work with our consumer code.

- Can anyone suggest references or even rough suggestions to do this refactor?  Even something like "move your code using channels.binding to a subclass of such-and-such and dispatch it upon so-and-so event," would really help! 
- Can someone with more experience opine whether [channels_binding](https://pypi.org/project/channels-binding/) would make an effective replacement?

**Comment by carltongibson** [comment]
Hi @TW4177 I'm afraid I don't have an instant answer for you, but there are several projects which turn up here that might show the way forward: https://github.com/search?q=django+channels+binding&type=repositories
